include/sharg/detail/format_base.hpp

```diff
@@ -213,65 +214,61 @@ class format_help_base : public format_base
+        std::string info{config.description};
+        info += ((config.required) ? std::string{" "} : detail::to_string(" Default: ", value, ". "));
+        info += config.validator.get_help_page_message();
         store_help_page_element(
             [this, id, info]()
             {
                 derived_t().print_list_item(id, info);
             },
-            spec);
+            config.advanced,
+            config.hidden);
     }
 
     /*!\brief Adds a sharg::print_list_item call to be evaluated later on.
      * \copydetails sharg::parser::add_flag
      */
-    void add_flag(bool & SHARG_DOXYGEN_ONLY(value),
-                  char const short_id,
-                  std::string const & long_id,
-                  std::string const & desc,
-                  option_spec const spec)
+    template <typename config_type>
+    void add_flag(bool & SHARG_DOXYGEN_ONLY(value), config_type const & config)
     {
-        std::string id = prep_id_for_help(short_id, long_id);
+        std::string id = prep_id_for_help(config.short_id, config.long_id);
         store_help_page_element(
-            [this, id, desc]()
+            [this, id, config]()
             {
-                derived_t().print_list_item(id, desc);
+                derived_t().print_list_item(id, config.description);
             },
-            spec);
+            config.advanced,
+            config.hidden);
     }
 
     /*!\brief Adds a sharg::print_list_item call to be evaluated later on.
      * \copydetails sharg::parser::add_positional_option
      */
-    template <typename option_type, typename validator_type>
-    void add_positional_option(option_type & value, std::string const & desc, validator_type & option_validator)
+    template <typename option_type, typename config_type>
+    void add_positional_option(option_type & value, config_type const & config)
     {
-        std::string msg = option_validator.get_help_page_message();
+        std::string msg = config.validator.get_help_page_message();
 
-        // clang-format off
-        positional_option_calls.push_back([this, &value, desc, msg] ()
-        {
-            ++positional_option_count;
-            derived_t().print_list_item(detail::to_string("\\fBARGUMENT-", positional_option_count, "\\fP ",
-                                                          option_type_and_list_info(value)),
-                                        desc +
-                                        // a list at the end may be empty and thus have a default value
-                                        ((detail::is_container_option<option_type>)
-                                            ? detail::to_string(" Default: ", value, ". ")
-                                            : std::string{" "}) +
-                                        msg);
-        });
+        positional_option_calls.push_back(
+            [this, &value, config, msg]()
```

Comment:
I guess `config` is a copy because it's usually a temporary only used in the function call and the reference would be dangling afterward?

I guess we could move it `[this, &value, config = std::move(config), msg]` but it's not important :D

Reply:
> I guess config is a copy because it's usually a temporary only used in the function call and the reference would be dangling afterward?

exactly :)

> I guess we could move it [this, &value, config = std::move(config), msg] but it's not important :D

good idea

Reply:
on second thought, what if the user of positional options reuses a sharg config object? Passing a const & that we move.. not likely but still unintuitive?